test/integration/quic_http_integration_test.cc

```diff
@@ -663,8 +663,7 @@ TEST_P(QuicHttpIntegrationTest, PortMigrationFailureOnPathDegrading) {
   EXPECT_EQ(1024u * 2, upstream_request_->bodyLength());
 }
 
-// TODO(#19006): Track down the flakiness and re-enable.
-TEST_P(QuicHttpIntegrationTest, DISABLED_AdminDrainDrainsListeners) {
+TEST_P(QuicHttpIntegrationTest, AdminDrainDrainsListeners) {
```

Comment:
Woo hoo!

Would it make sense to try to write a unit test for the new logic in source/server/listener_manager_impl.cc?

Reply:
added